docs/troubleshooting.md

```diff
@@ -0,0 +1,94 @@
+# Troubleshooting
+
+## Common issue #1
+
+```
+fatal: [host.example.com]: UNREACHABLE! => {“changed”: false, “msg”: “Failed to connect to the host via ssh: Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).“, “unreachable”: true}
+```
+
+or
+
+```
+fatal: [host.example.com]: UNREACHABLE! => {"changed": false, "msg": "SSH password was requested, but none specified", "unreachable": true}
+```
+
+**Problem**: 
+
+  - Passwordless SSH connectivity is not correctly set up between your Ansible server and the remote servers in the inventory. 
+
+**Solutions**:
+
+  - _(recommended)_ Create an SSH key pair for the Ansible server and ensure the public key is placed in all the servers' `authorized_keys` file for the connecting user, e.g. `/root/.ssh/authorized_keys` or `/home/<user>/.ssh/authorized_keys`
+
+  - _(alternative)_ Provide a password for SSH connectivity during the playbook run:
+
+    ```bash
+    $ ansible-playbook --ask-pass ...other args
+    ```
+
+## Common issue #2
+
+```
+fatal: [host.example.com]: UNREACHABLE! => {"changed": false, "msg": "Host key checking is enabled, and SSH reported an unrecognized or mismatching host key.", "unreachable": true}
+```
+
+**Problem**: 
+
+  - SSH host key checking is enabled (Ansible default) but the servers you are trying to connect to are not present in `known_hosts` (or their key has changed)
+
+**Solutions**:
+
+- _(easiest, but least secure)_ Disable Ansible host key checking. In your Ansible configuration file, enter the following:
+
+  ```ini
+  [defaults]
+  host_key_checking = False
+  ```
+
+- _(annoying, but more secure)_ Connect to each of your servers manually from the Ansible host once, so that the `known_hosts` file becomes populated.
```

Comment:
Note: This is disabled by default in Cloudera-Deploy for the benefit of the novice user

Reply:
Noted